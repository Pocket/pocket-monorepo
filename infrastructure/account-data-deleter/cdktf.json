{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/main.js",
  "projectId": "90437d3d-e067-4603-8eda-11e297370139"
}