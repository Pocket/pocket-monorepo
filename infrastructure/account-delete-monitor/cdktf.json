{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/main.js",
  "projectId": "d9efd22b-fec2-4561-82e1-9319ba965703"
}