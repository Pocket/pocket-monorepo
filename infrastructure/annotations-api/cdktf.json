{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/main.js",
  "projectId": "5d96b3e5-a2f4-4187-a9a3-2356401e3092"
}