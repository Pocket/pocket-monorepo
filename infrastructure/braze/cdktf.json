{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/main.js",
  "projectId": "d67fe567-bd64-4c5c-b71b-708889a29bb2"
}