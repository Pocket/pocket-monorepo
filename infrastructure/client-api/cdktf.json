{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/main.js",
  "projectId": "7e0b548c-601b-4fd5-9387-cd9c77f5007c"
}
