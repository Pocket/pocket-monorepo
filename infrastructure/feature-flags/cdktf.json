{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/main.js",
  "projectId": "0dfd58e5-3d74-43bd-86fe-fafe473e213e"
}
