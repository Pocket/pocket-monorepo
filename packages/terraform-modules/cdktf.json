{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/example.js",
  "projectId": "67d4fb1a-0682-48af-bc23-a5ddc7941efc"
}