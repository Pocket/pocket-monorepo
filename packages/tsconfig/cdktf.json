{
    "$schema": "https://json.schemastore.org/tsconfig",
    "compilerOptions": {
      "experimentalDecorators": true,
      "emitDecoratorMetadata": true,
      "rewriteRelativeImportExtensions": true,
      "allowImportingTsExtensions": true,
      "target": "ES2024",
      "module": "NodeNext",
      "moduleResolution": "NodeNext",
      "removeComments": true,
      "esModuleInterop": true,
      "noEmitOnError": false,
      "sourceMap": true,
      "inlineSources": true,
      "sourceRoot": "/",
      "declaration": true,
      "strictNullChecks": true
    }
  }