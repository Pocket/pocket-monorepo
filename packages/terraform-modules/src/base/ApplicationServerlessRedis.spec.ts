import { Testing } from 'cdktf';

import {
  ApplicationServerlessRedis,
  ApplicationServerlessRedisProps,
} from './ApplicationServerlessRedis';

const BASE_CONFIG: ApplicationServerlessRedisProps = {
  allowedIngressSecurityGroupIds: [],
  subnetIds: ['subnet-1', 'subnet-2', 'subnet-3'],
  vpcId: 'cool-vpc',
  prefix: `abides-dev`,
};

describe('ApplicationRedis', () => {
  it('renders redis with minimal config', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationServerlessRedis(stack, 'testRedis', BASE_CONFIG);
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders redis with tags', () => {
    const config: ApplicationServerlessRedisProps = {
      ...BASE_CONFIG,
      tags: {
        letsgo: 'bowling',
        donnie: 'throwinrockstonight',
      },
    };
    const synthed = Testing.synthScope((stack) => {
      new ApplicationServerlessRedis(stack, 'testRedis', config);
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders redis with more then 3 subnets', () => {
    const config: ApplicationServerlessRedisProps = {
      ...BASE_CONFIG,
      subnetIds: ['subnet-1', 'subnet-2', 'subnet-3', 'subnet-4'],
    };
    const synthed = Testing.synthScope((stack) => {
      new ApplicationServerlessRedis(stack, 'testRedis', config);
    });
    expect(synthed).toMatchSnapshot();
  });

  it('errors redis with less then 2 subnets', () => {
    const config: ApplicationServerlessRedisProps = {
      ...BASE_CONFIG,
      subnetIds: ['subnet-1'],
    };

    Testing.synthScope((stack) => {
      expect(() => {
        new ApplicationServerlessRedis(stack, 'testRedis', config);
      }).toThrow('Elasticache serverless requires at least 2 subnets');
    });
  });
});
