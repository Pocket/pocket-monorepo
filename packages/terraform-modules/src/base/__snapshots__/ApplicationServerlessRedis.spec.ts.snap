// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationRedis renders redis with minimal config 1`] = `
"{
  "data": {
    "aws_vpc": {
      "testRedis_vpc_442CAD26": {
        "filter": [
          {
            "name": "vpc-id",
            "values": [
              "cool-vpc"
            ]
          }
        ]
      }
    }
  },
  "resource": {
    "aws_elasticache_serverless_cache": {
      "testRedis_elasticache_serverless_E5B11082": {
        "description": "Redis for abides-dev",
        "engine": "redis",
        "name": "abides-dev",
        "security_group_ids": [
          "\${aws_security_group.testRedis_elasticache_security_group_36C6CB6D.id}"
        ],
        "subnet_ids": [
          "subnet-1",
          "subnet-2",
          "subnet-3"
        ]
      }
    },
    "aws_security_group": {
      "testRedis_elasticache_security_group_36C6CB6D": {
        "description": "Managed by Terraform",
        "ingress": [
          {
            "cidr_blocks": null,
            "description": null,
            "from_port": 6379,
            "ipv6_cidr_blocks": null,
            "prefix_list_ids": null,
            "protocol": "tcp",
            "security_groups": [
            ],
            "self": null,
            "to_port": 6379
          }
        ],
        "name_prefix": "abides-dev",
        "vpc_id": "\${data.aws_vpc.testRedis_vpc_442CAD26.id}"
      }
    }
  }
}"
`;

exports[`ApplicationRedis renders redis with more then 3 subnets 1`] = `
"{
  "data": {
    "aws_vpc": {
      "testRedis_vpc_442CAD26": {
        "filter": [
          {
            "name": "vpc-id",
            "values": [
              "cool-vpc"
            ]
          }
        ]
      }
    }
  },
  "resource": {
    "aws_elasticache_serverless_cache": {
      "testRedis_elasticache_serverless_E5B11082": {
        "description": "Redis for abides-dev",
        "engine": "redis",
        "name": "abides-dev",
        "security_group_ids": [
          "\${aws_security_group.testRedis_elasticache_security_group_36C6CB6D.id}"
        ],
        "subnet_ids": [
          "subnet-1",
          "subnet-2",
          "subnet-3"
        ]
      }
    },
    "aws_security_group": {
      "testRedis_elasticache_security_group_36C6CB6D": {
        "description": "Managed by Terraform",
        "ingress": [
          {
            "cidr_blocks": null,
            "description": null,
            "from_port": 6379,
            "ipv6_cidr_blocks": null,
            "prefix_list_ids": null,
            "protocol": "tcp",
            "security_groups": [
            ],
            "self": null,
            "to_port": 6379
          }
        ],
        "name_prefix": "abides-dev",
        "vpc_id": "\${data.aws_vpc.testRedis_vpc_442CAD26.id}"
      }
    }
  }
}"
`;

exports[`ApplicationRedis renders redis with tags 1`] = `
"{
  "data": {
    "aws_vpc": {
      "testRedis_vpc_442CAD26": {
        "filter": [
          {
            "name": "vpc-id",
            "values": [
              "cool-vpc"
            ]
          }
        ]
      }
    }
  },
  "resource": {
    "aws_elasticache_serverless_cache": {
      "testRedis_elasticache_serverless_E5B11082": {
        "description": "Redis for abides-dev",
        "engine": "redis",
        "name": "abides-dev",
        "security_group_ids": [
          "\${aws_security_group.testRedis_elasticache_security_group_36C6CB6D.id}"
        ],
        "subnet_ids": [
          "subnet-1",
          "subnet-2",
          "subnet-3"
        ],
        "tags": {
          "donnie": "throwinrockstonight",
          "letsgo": "bowling"
        }
      }
    },
    "aws_security_group": {
      "testRedis_elasticache_security_group_36C6CB6D": {
        "description": "Managed by Terraform",
        "ingress": [
          {
            "cidr_blocks": null,
            "description": null,
            "from_port": 6379,
            "ipv6_cidr_blocks": null,
            "prefix_list_ids": null,
            "protocol": "tcp",
            "security_groups": [
            ],
            "self": null,
            "to_port": 6379
          }
        ],
        "name_prefix": "abides-dev",
        "tags": {
          "donnie": "throwinrockstonight",
          "letsgo": "bowling"
        },
        "vpc_id": "\${data.aws_vpc.testRedis_vpc_442CAD26.id}"
      }
    }
  }
}"
`;
